api/src/org/apache/cloudstack/api/command/user/snapshot/ListSnapshotsCmd.java

```diff
@@ -115,8 +115,10 @@ public void execute() {
         List<SnapshotResponse> snapshotResponses = new ArrayList<SnapshotResponse>();
         for (Snapshot snapshot : result.first()) {
             SnapshotResponse snapshotResponse = _responseGenerator.createSnapshotResponse(snapshot);
-            snapshotResponse.setObjectName("snapshot");
-            snapshotResponses.add(snapshotResponse);
+            if (snapshotResponse != null) {
+                snapshotResponse.setObjectName("snapshot");
```

Comment:
This check is added as we introduced a case in `createSnapshotResponse` null is returned, and in that case we should skip the response